### PR TITLE
tests: Add descriptorBufferAddressSpaceSize

### PR DESCRIPTION
**Pull request by spencer-lunarg**
tests to check https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/5732

### CODE REVIEW

File: layers/core_checks/cc_buffer.cpp [hunk 1]
```diff
@@ -211,10 +211,8 @@ bool CoreChecks::PreCallValidateCreateBuffer(VkDevice device, const VkBufferCrea
                 ") is greater than specified in properties field samplerDescriptorBufferAddressSpaceSize (%" PRIuLEAST64 ").",
                 pCreateInfo->size, samplerDescriptorBufferAddressSpaceSize.load(),
                 phys_dev_ext_props.descriptor_buffer_props.samplerDescriptorBufferAddressSpaceSize);
-        }
-
-        if (pCreateInfo->size + descriptorBufferAddressSpaceSize >
-            phys_dev_ext_props.descriptor_buffer_props.descriptorBufferAddressSpaceSize) {
+        } else if (pCreateInfo->size + descriptorBufferAddressSpaceSize >
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_buffer.cpp:214]: there was no reason to print 2 errors for same VUID here

**Review by arno-lunarg — APPROVED** [review]
LGTM, but is this just a draft then?

File: tests/negative/descriptor_buffer.cpp [hunk 2]
```diff
@@ -1499,3 +1499,51 @@ TEST_F(VkLayerTest, DescriptorBufferInvalidExtensionCombination) {
...
+    ASSERT_NO_FATAL_FAILURE(InitState(nullptr, &descriptor_buffer_features));
+
+    auto descriptor_buffer_properties = LvlInitStruct<VkPhysicalDeviceDescriptorBufferPropertiesEXT>();
+    GetPhysicalDeviceProperties2(descriptor_buffer_properties);
+    // After a few GB, can have memory issues running these tests
+    // descriptorBufferAddressSpaceSize is always the largest of the 3 buffer address size limits
+    const VkDeviceSize max_limit = 1 << 31;
+    if (descriptor_buffer_properties.descriptorBufferAddressSpaceSize > max_limit) {
+        GTEST_SKIP() << "descriptorBufferAddressSpaceSize are too large";
+    }
```
Comment (juan-lunarg) [inline tests/negative/descriptor_buffer.cpp:1527]: Should we run this test on Jenkins CI? Is the MockICD enough?

**Review by juan-lunarg — COMMENTED** [review]
I ran `VkPositiveLayerTest.DescriptorBufferSetCreateBuffer` without your changes to repro the false positive. But the test passes fine.

Looks like this doesn't address the original false positive? Does the user have a false positive?

File: tests/negative/descriptor_buffer.cpp (hunk 2 above)
Comment (spencer-lunarg) [reply tests/negative/descriptor_buffer.cpp:1527]: MockICD and local GPU both worked

**Comment by spencer-lunarg** [comment]
This doesn't "fix" the issue, but the issue is because `descriptorBufferAddressSpaceSize` is returning `0` which not sure, but we have tests now regardless to prove it "should" work

**Comment by dustContributor** [comment]
I closed my issue #5732 It wasn't a false positive, it was PEBCAK. I wasn't passing the VK_EXT_descriptor_buffer extension to the device creation. Thank you for your time.

**Pull request merged**